cloudinit/sources/DataSourceOracle.py

```diff
@@ -167,9 +174,16 @@ def _unpickle(self, ci_pkl_version: int) -> None:
             )
         if not hasattr(self, "_network_config"):
             self._network_config = {"config": [], "version": 1}
+        if not hasattr(self, "_network_config_v1"):
+            self._network_config_v1 = {"config": [], "version": 1}
 
     def _has_network_config(self) -> bool:
-        return bool(self._network_config.get("config", []))
+        if self._network_config:
```

Comment:
Nit: I think this whole section could be `return "config" in self._network_config or "ethernets" in self._network_config`